### PR TITLE
Backend work for Notebook Parameterization and Tests

### PR DESCRIPTION
**Pull request by VasuBhog**
This PR introduces the backend cell model work that is required for Parameterization. There are two important tags for parameterization as a whole: **parameters** and **injected-parameters**. This enables the use of [Papermill](https://github.com/nteract/papermill) to work properly with ADS Notebooks. 

The current backend work is demonstrated in the gif below. It focuses on using the cell toolbar action to change the metadata based on the action of "_Make parameter cell_" and "_Remove parameter cell_". I have added tests for both the parameter cell as well as the injected-parameter cell. 
The injected-parameter cell will be added to a notebook below the parameters cell when a user uses Papermill (for example) to parameterize a notebook. 

Please leave any feedback or questions!

**Work to do**: UI components for the visual tag of the parameter cell. 

parameterization_backend
Gif: Click cell -> Use toolbar action 'Make parameter cell' -> Save notebook -> View metadata of notebook cell with new tag "parameters" -> Use toolbar action 'Remove parameter cell' -> Save notebook -> View metadata of notebook cell and see that the "parameters" tag is no longer there.

### CODE REVIEW

**Review by lucyzhang929 — APPROVED** [review]

File: src/sql/workbench/contrib/notebook/browser/cellToolbarActions.ts [hunk 1]
```diff
@@ -364,3 +367,41 @@ export class ToggleMoreActions extends Action {
...
+				}
+			}
+		} catch (error) {
+			let message = getErrorMessage(error);
+			this.notificationService.notify({
+				severity: Severity.Error,
+				message: message
+			});
+		}
+		return Promise.resolve();
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/browser/cellToolbarActions.ts:405]: Just remove this - not needed in async functions

File: src/sql/workbench/contrib/notebook/browser/cellToolbarActions.ts [hunk 2]
```diff
@@ -364,3 +367,41 @@ export class ToggleMoreActions extends Action {
...
+	constructor(id: string,
+		label: string,
+		private parametersCell: boolean,
+		@INotificationService notificationService: INotificationService
+	) {
+		super(id, label, undefined, notificationService);
+	}
+
+	public canRun(context: CellContext): boolean {
+		return context.cell && context.cell.cellType === CellTypes.Code;
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/browser/cellToolbarActions.ts:381]: `context.cell?.cellType === CellTypes.Code`

File: src/sql/workbench/contrib/notebook/test/electron-browser/cell.test.ts [hunk 3]
```diff
@@ -349,11 +349,16 @@ suite('Cell Model', function (): void {
...
 		assert(cell.isCollapsed === false);
+		assert(cell.isParameter === false);
+
 		cell.isCollapsed = true;
+		cell.isParameter = true;
 		// The typescript compiler will complain if we don't ignore the error from the following line,
 		// claiming that cell.isCollapsed will return true. It doesn't.
 		// @ts-ignore
 		assert(cell.isCollapsed === false);
+		// @ts-ignore
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/notebook/test/electron-browser/cell.test.ts:360]: Is the ts-ignore necessary here too?

File: src/sql/workbench/services/notebook/browser/models/cell.ts [hunk 4]
```diff
@@ -340,6 +343,76 @@ export class CellModel extends Disposable implements ICellModel {
...
+			}
+		}
+
+		if (stateChanged) {
+			this._onParameterStateChanged.fire(this._isParameter);
+			this.sendChangeToNotebook(NotebookChangeType.CellInputVisibilityChanged);
+		}
+	}
+
+	// Injected Parameters will be used for future scenarios
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/models/cell.ts:385]: Use XML doc format - /** */

**Review by Charles-Gagnon — APPROVED** [review]

**Review by halerankin — APPROVED** [review]

File: src/sql/workbench/contrib/notebook/test/electron-browser/cell.test.ts (hunk 3 above)
Comment (VasuBhog) [reply src/sql/workbench/contrib/notebook/test/electron-browser/cell.test.ts:360]: Yes, the same problem occurs for cell.isCollapsed. I can edit the comment to inform both cell.isCollapsed and cell.isParamterer.

**Review by corivera — APPROVED** [review]

**Review by barbaravaldez — APPROVED** [review]

**Review by chlafreniere — APPROVED** [review]

**Pull request merged**